cli/tests/test_fix_command.rs

```diff
@@ -0,0 +1,513 @@
+    insta::assert_snapshot!(stderr, @r###"
+    Fixed 1 commits of 1 checked.
+    Working copy now at: mzvwutvl 43aa7c45 (no description set)
+    Parent commit      : qpvuntsm 34782c48 a | (no description set)
+    Parent commit      : kkmpptxz 82e9bc6a b | (no description set)
+    Added 0 files, modified 4 files, removed 0 files
+    "###);
+    let content = test_env.jj_cmd_success(&repo_path, &["print", "file_a", "-r", "@"]);
+    insta::assert_snapshot!(content, @"a live");
+    let content = test_env.jj_cmd_success(&repo_path, &["print", "file_b", "-r", "@"]);
+    insta::assert_snapshot!(content, @"b live");
+    let content = test_env.jj_cmd_success(&repo_path, &["print", "file_c", "-r", "@"]);
+    insta::assert_snapshot!(content, @"c live");
+    let content = test_env.jj_cmd_success(&repo_path, &["print", "file_d", "-r", "@"]);
+    insta::assert_snapshot!(content, @"d live");
+}
+
+#[test]
+fn test_fix_both_sides_of_conflict() {
+    let (test_env, repo_path) = init_with_fake_formatter("--reverse");
+    std::fs::write(repo_path.join("file"), "content a\n").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["branch", "create", "a"]);
+    test_env.jj_cmd_ok(&repo_path, &["new", "@-"]);
+    std::fs::write(repo_path.join("file"), "content b\n").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["branch", "create", "b"]);
+    test_env.jj_cmd_ok(&repo_path, &["new", "a", "b"]);
+
+    // The conflicts are not different from the merged parent, so they would not be
+    // fixed if we didn't fix the parents also.
+    let (stdout, stderr) = test_env.jj_cmd_ok(&repo_path, &["fix", "-s", "a", "-s", "b"]);
+    insta::assert_snapshot!(stdout, @"");
+    insta::assert_snapshot!(stderr, @r###"
+    Fixed 3 commits of 3 checked.
+    Working copy now at: mzvwutvl 36064b9c (conflict) (empty) (no description set)
+    Parent commit      : qpvuntsm eb631ae2 a | (no description set)
+    Parent commit      : kkmpptxz 9c4b133b b | (no description set)
+    Added 0 files, modified 1 files, removed 0 files
+    There are unresolved conflicts at these paths:
+    file    2-sided conflict
+    "###);
+    let content = test_env.jj_cmd_success(&repo_path, &["print", "file", "-r", "a"]);
+    insta::assert_snapshot!(content, @r###"
+    a tnetnoc
+    "###);
+    let content = test_env.jj_cmd_success(&repo_path, &["print", "file", "-r", "b"]);
+    insta::assert_snapshot!(content, @r###"
+    b tnetnoc
+    "###);
+    let content = test_env.jj_cmd_success(&repo_path, &["print", "file", "-r", "@"]);
+    insta::assert_snapshot!(content, @r###"
+    <<<<<<< Conflict 1 of 1
+    %%%%%%% Changes from base to side #1
+    +a tnetnoc
+    +++++++ Contents of side #2
+    b tnetnoc
+    >>>>>>>
+    "###);
+}
+
+#[test]
+fn test_fix_resolve_conflict() {
+    // If both sides of the conflict look the same after being fixed, the conflict
+    // will be resolved.
+    let (test_env, repo_path) = init_with_fake_formatter("--stdout 'new content'");
```

Comment:
If the fake formatter's default behavior was to uppercase, I think you could remove the `--stdout` argument from it (at least here?) and maybe produce more interesting test cases too